ts/types/Attachment.ts

```diff
@@ -34,3 +43,59 @@ export const isVisualMedia = (attachment: Attachment): boolean => {
   const isSupportedVideoType = GoogleChrome.isVideoTypeSupported(contentType);
   return isSupportedImageType || isSupportedVideoType;
 };
+
+export const save = ({
+  attachment,
+  document,
+  getAbsolutePath,
+  timestamp,
+}: {
+  attachment: Attachment;
+  document: Document;
+  getAbsolutePath: (relativePath: string) => string;
+  timestamp?: number;
+}): void => {
+  const url = !is.undefined(attachment.path)
+    ? getAbsolutePath(attachment.path)
+    : arrayBufferToObjectURL({
+        data: attachment.data,
+        type: SAVE_CONTENT_TYPE,
+      });
+  const filename = getSuggestedFilename({ attachment, timestamp });
+  saveURLAsFile({ url, filename, document });
+  URL.revokeObjectURL(url);
+};
+
+export const getSuggestedFilename = ({
+  attachment,
+  timestamp,
+}: {
+  attachment: Attachment;
+  timestamp?: number | Date;
+}): string => {
+  if (attachment.fileName) {
+    return attachment.fileName;
+  }
+
+  const prefix = 'signal-attachment';
+  const suffix = timestamp
+    ? moment(timestamp).format('-YYYY-MM-DD-HHmmss')
+    : '';
+  const fileType = getFileExtension(attachment);
+  const extension = fileType ? `.${fileType}` : '';
+  return `${prefix}${suffix}${extension}`;
+};
+
+export const getFileExtension = (attachment: Attachment): string | null => {
+  if (!attachment.contentType) {
+    return null;
+  }
+
+  switch (attachment.contentType) {
+    case 'video/quicktime':
+      return 'mov';
+    default:
+      // TODO: Use better MIME --> file extension mapping:
```

Comment:
Is there an issue created for this somewhere?

Reply:
Yes, created one in Trello.